### PR TITLE
Fix: Start Wasm table for the RTS at offset >= 1

### PR DESCRIPTION
**Pull request by luc-blaeser**
Rust requires a table offset of at least 1 as the element index 0 is considered invalid and causes a debug null check to panic when called. On the other hand, `elem[0]` can be used by the Motoko backend code, as correct Rust-generated Wasm code does not call `elem[0]`.

This bug is independent of `wasm32` and `wasm64`.

This issue has been observed in https://github.com/dfinity/motoko/actions/runs/10703077671/job/29672766216?pr=4683 and happened only on Linux and only under `nix-build` (not `nix-shell`).

### CODE REVIEW

File: src/linking/linkModule.ml [hunk 1]
```diff
@@ -846,7 +846,13 @@ let link (em1 : extended_module) libname (em2 : extended_module) =
     )
   end else ();
 
-  let old_table_size = read_table_size em1.module_ in
+  let max x y = if x >= y then x else y in
```
Comment (ggreif) [inline src/linking/linkModule.ml:849]:
```suggestion
  let max x y = if x >= y then x else y in (* use `Int.max` when bumping to 4.13 *)
```

**Review by ggreif — COMMENTED** [review]
Why didn't the bytecode-verifier in the replica catch this? Do we have Motoko-compiled canisters out there with this bug baked-in?

File: src/linking/linkModule.ml (hunk 1 above)
Comment (luc-blaeser) [reply src/linking/linkModule.ml:849]: Thank you. I also noticed it would be in Ocaml 4.13. Good to have a todo comment in the code.

**Comment by luc-blaeser** [comment]
> Why didn't the bytecode-verifier in the replica catch this? Do we have Motoko-compiled canisters out there with this bug baked-in?

It is actually only a requirement by Rust, not Wasm. It only affects debug Rust compilation AFAIK. I believe people probably do not release debug Motoko code. So, probably it is okay.

**Review by crusso — APPROVED** [review]
LGTM

**Review by ggreif — APPROVED** [review]
Fine by me!

**Pull request merged**